src/Syntax.hs

```diff
@@ -31,7 +30,7 @@ makeLenses ''Stx
 data ExprF a
   = Id Text
   | String Text
-  | Sig Signal
+  | LitInt Integer
```

Comment:
For consistency, I think we should either use `String` and `Int` or `LitString` and `LitInt`.

Reply:
I'd be fine with tackling that in a follow-up. I chose `LitInt` to avoid clashing with Haskell's built-in `Integer`.

Reply:
What was the clash? `String` is, of course, also a built-in type, but it's a type, so it doesn't conflict with `ExprF`'s `String` data constructor. I would have expected the same to happen with `Integer`.

Reply:
@gelisam Yep, I just have a stylistic preference to avoid reusing common built-in names, because it can be confusing.